fhevm/precompiles.go

```diff
@@ -2069,58 +2064,25 @@ func reencryptRun(environment EVMEnvironment, caller common.Address, addr common
 	ct := getVerifiedCiphertext(environment, common.BytesToHash(input[0:32]))
 	if ct != nil {
 		// Make sure we don't decrypt before any optimistic requires are checked.
-		// optReqResult, optReqErr := evaluateRemainingOptimisticRequires(environment)
-		// if optReqErr != nil {
-		// 	return nil, optReqErr
-		// } else if !optReqResult {
-		// 	return nil, ErrExecutionReverted
-		// }
-
-		var fheType kms.FheType
-		switch ct.ciphertext.fheUintType {
-		case FheUint8:
-			fheType = kms.FheType_Euint8
-		case FheUint16:
-			fheType = kms.FheType_Euint16
-		case FheUint32:
-			fheType = kms.FheType_Euint32
+		optReqResult, optReqErr := evaluateRemainingOptimisticRequires(environment)
+		if optReqErr != nil {
+			return nil, optReqErr
+		} else if !optReqResult {
+			return nil, ErrExecutionReverted
```

Comment:
These 4 lines are commented in Zama's new fhevm-go, but not commented in their old go-ethereum fork [here](https://github.com/zama-ai/go-ethereum/blob/f24d3e742ec643b848e69b6daab671dc787f7d5d/core/vm/contracts.go#L2053-L2058).

Any idea if we should keep them?

Reply:
I don't think it would be a bad thing to keep it.